src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs

```diff
@@ -102,27 +104,24 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
         // Create a simple interval tree for simplification spans.
         var spansTree = new TextSpanIntervalTree(spans);
 
-        bool isNodeOrTokenOutsideSimplifySpans(SyntaxNodeOrToken nodeOrToken)
-            => !spansTree.HasIntervalThatOverlapsWith(nodeOrToken.FullSpan.Start, nodeOrToken.FullSpan.Length);
-
         var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-        var root = await semanticModel.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+        var root = (TCompilationUnitSyntax)await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
         // prep namespace imports marked for simplification 
         var removeIfUnusedAnnotation = new SyntaxAnnotation();
         var originalRoot = root;
-        root = PrepareNamespaceImportsForRemovalIfUnused(document, root, removeIfUnusedAnnotation, isNodeOrTokenOutsideSimplifySpans);
+        root = PrepareNamespaceImportsForRemovalIfUnused(root, removeIfUnusedAnnotation, IsNodeOrTokenOutsideSimplifySpans);
         var hasImportsToSimplify = root != originalRoot;
 
         if (hasImportsToSimplify)
         {
             document = document.WithSyntaxRoot(root);
             semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
>semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false); [](http://example.com/codeflow?start=12&length=102)

can this be moved to after the if so it doesn't need to be done twice?

Reply:
actually, can it be moved to down by where the ReduceAsync call is made?

Reply:
yup. done.